File: leetcode/2090.k-radius-subarray-averages/solution.cpp
#include "../../common.hpp"
// https://leetcode.com/problems/k-radius-subarray-averages

auto fastio = []() {
  std::ios::sync_with_stdio(false);
  std::cin.tie(nullptr);
  std::cout.tie(nullptr);
  return nullptr;
}();

class Solution {
public:
  static vector<int> getAverages(vector<int> &nums, int k) {
    if (k == 0)
      return std::move(nums);
    int n = nums.size() - 2 * k;
    if (n <= 0)
      return vector<int>(nums.size(), -1);
    vector<int> res;
    res.reserve(nums.size());
    res.resize(k, -1);
    auto beg = nums.begin(), end = beg + 2 * k;
    auto sum = std::accumulate(beg, end, 0ll);
    for (int i = 0; i < n; ++i) {
      sum += *end++;
      res.push_back(sum / (2 * k + 1));
      sum -= *beg++;
    }
    res.resize(nums.size(), -1);
    return res;
  }
};

int main(int argc, char *argv[]) {
  Solution s;
  return 0;
}

File: leetcode/2090.k-radius-subarray-averages/solution.rs
struct Solution;
// https://leetcode.com/problems/k-radius-subarray-averages
impl Solution {
    pub fn get_averages(nums: Vec<i32>, k: i32) -> Vec<i32> {
        let len = nums.len();
        if k == 0 {
            nums
        } else if len as i32 <= 2 * k {
            vec![-1; len]
        } else {
            let mut result = Vec::with_capacity(nums.len());
            let mut sum = 0u64;
            let n = (2 * k + 1) as u64;
            let mut it = nums.iter();
            for _ in 0..k {
                result.push(-1)
            }
            for _ in 0..2 * k {
                sum += (*it.next().unwrap()) as u64;
            }
            for (&x, &y) in nums.iter().zip(it) {
                sum += y as u64;
                result.push((sum / n) as i32);
                sum -= x as u64;
            }
            for _ in 0..k {
                result.push(-1)
            }
            result
        }
    }
}
